pypesto/store/save_to_hdf5.py

```diff
@@ -145,3 +145,121 @@ def write(self, result: Result, overwrite=False):
+                                        "contains information about "
+                                        "optimization result. If you wish "
+                                        "to overwrite it, set "
+                                        "overwrite=True.")
+            for key in result.sample_result.keys():
+                if isinstance(result.sample_result[key], np.ndarray):
+                    write_float_array(results_grp,
+                                      key,
+                                      result.sample_result[key])
+                elif result.sample_result[key] is not None:
+                    results_grp.attrs[key] = result.sample_result[key]
+            f.flush()
+
+
+class ProfileResultHDF5Writer():
+    """
+    Writer of the HDF5 result files.
+
+    Attributes
+    -------------
+    storage_filename:
+        HDF5 result file name
+    """
+
+    def __init__(self, storage_filename: str):
+        """
+        Parameters
+        ----------
+
+        storage_filename: str
+            HDF5 result file name
+        """
+        self.storage_filename = storage_filename
+
+    def write(self, result: Result, overwrite=False):
+        """
+        Write HDF5 result file from pyPESTO result object.
+        """
+
+        # Create destination directory
+        if isinstance(self.storage_filename, str):
+            basedir = os.path.dirname(self.storage_filename)
+            if basedir:
+                os.makedirs(basedir, exist_ok=True)
+
+        with h5py.File(self.storage_filename, "a") as f:
+            profiling_grp = get_or_create_group(f, "profiling")
+
+            for profile_id, profile in enumerate(result.profile_result.list):
+                profile_grp = get_or_create_group(profiling_grp,
+                                                  str(profile_id))
+                for parameter_id, parameter_profile in enumerate(profile):
+                    result_grp = get_or_create_group(profile_grp,
+                                                     str(parameter_id))
+                    if parameter_profile is None:
+                        result_grp.attrs['IsNone'] = True
+                    else:
+                        result_grp.attrs['IsNone'] = False
+                        if not overwrite:
+                            for key in parameter_profile.keys():
+                                if (key in result_grp.keys()
+                                        or key in result_grp.attrs):
+                                    raise Exception("The file already exists"
+                                                    " and contains information"
+                                                    " about optimization"
+                                                    " result. If you wish"
+                                                    " to overwrite it, set"
+                                                    " overwrite=True.")
```

Comment:
Reduce nesting depth. Also seems to be redundant with some function further up. Move to separate function?

Reply:
Reduced the nesting depth by one.

Reply:
added a check_overwrite function for this, which now also deletes the files if overwrite=True and the files exist. Which solves the bug i mentioned in the initial PR.